### PR TITLE
[Dynamo] modify IPEX backend

### PR DESCRIPTION
**Pull request by jiayisunx**
1. Combine the two backends ‘ipex_fp32’ and ‘ipex_bf16’ into one backend ‘ipex’.
2. Modify IPEX backend to work in fake mode and symbolic mode.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @desertfire

### CODE REVIEW

File: test/dynamo/test_optimizations.py [hunk 1]
```diff
@@ -166,11 +165,15 @@ def test_ipex_fp32(self):
...
         input = torch.randn(8, 3, 64, 64).contiguous(memory_format=torch.channels_last)
         r1 = model(input)
-        opt_model = torch._dynamo.optimize(backends.ipex_fp32)(model)
-        with torch.no_grad():
-            r2 = opt_model(input)
-        self.assertTrue(same(r1, r2))
-        self.assertEqual(r2.dtype, torch.float32)
+        for dynamic_shapes in [True, False]:
+            torch._dynamo.reset()
+            torch._dynamo.config.dynamic_shapes = dynamic_shapes
```
Comment (jgong5) [inline test/dynamo/test_optimizations.py:170]: Can you use `with patch` to make sure the configuration changes here won't impact others?

File: torch/_dynamo/optimizations/backends.py [hunk 2]
```diff
@@ -788,14 +767,35 @@ def fwd(*args):
     return fwd
 
 
-def ipex_fp32(gm: torch.fx.GraphModule, example_inputs):
-    kwargs_ipex = {"datatype": "fp32"}
-    return BACKENDS["ipex"](gm, example_inputs, **kwargs_ipex)
+@create_backend
+def ipex(subgraph):
+    import intel_extension_for_pytorch  # type: ignore[import]  # noqa: F401
```
Comment (jgong5) [inline torch/_dynamo/optimizations/backends.py:772]: Do "try-catch" and fall back to eager on exception?

**Review by jgong5 — COMMENTED** [review]
Can you add PR description on why you made the changes and what you changed in brief?

File: test/dynamo/test_optimizations.py (hunk 1 above)
Comment (jiayisunx) [reply test/dynamo/test_optimizations.py]: Done.

File: torch/_dynamo/optimizations/backends.py (hunk 2 above)
Comment (jiayisunx) [reply torch/_dynamo/optimizations/backends.py:772]: I checked other backends like onnx backend, they also don't have fall back path. So I'm not sure if it needs to be added.

Comment (jgong5) [reply torch/_dynamo/optimizations/backends.py:772]: I saw the "tvm" backend does fallbacks. Anyway, at least, we'd better print a readable error message on import error?

Comment (jiayisunx) [reply torch/_dynamo/optimizations/backends.py]: have added a readable error message, thanks!

File: torch/_dynamo/optimizations/backends.py [hunk 3]
```diff
@@ -769,7 +769,11 @@ def fwd(*args):
 
 @create_backend
 def ipex(subgraph):
-    import intel_extension_for_pytorch  # type: ignore[import]  # noqa: F401
+    try:
+        import intel_extension_for_pytorch  # type: ignore[import]  # noqa: F401
+    except ImportError:
+        log.exception("ipex backend fails. Cannot import intel_extension_for_pytorch")
```
Comment (jgong5) [inline torch/_dynamo/optimizations/backends.py:775]: How about "Unable to import Intel Extension for PyTorch (IPEX). Please install the right version of IPEX that matches the PyTorch version being used. Refer to https://github.com/intel/intel-extension-for-pytorch for details."

Comment (jiayisunx) [reply torch/_dynamo/optimizations/backends.py]: It's better, thank you!

**Review by jgong5 — APPROVED** [review]

**Comment by jiayisunx** [comment]
@jansel, could you please help to review this PR? thanks!

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**